src/nvim/api/window.c

```diff
@@ -134,25 +153,32 @@ void nvim_win_set_cursor(Window window, ArrayOf(Integer, 2) pos, Error *err)
     return;
   }
 
-  win->w_cursor.lnum = (linenr_T)row;
-  win->w_cursor.col = (colnr_T)col;
-  win->w_cursor.coladd = 0;
-  // When column is out of range silently correct it.
-  check_cursor_col(win);
-
-  // Make sure we stick in this column.
-  win->w_set_curswant = true;
-
-  // make sure cursor is in visible range and
-  // cursorcolumn and cursorline are updated even if win != curwin
   switchwin_T switchwin;
-  switch_win(&switchwin, win, NULL, true);
-  update_topline(curwin);
-  validate_cursor(curwin);
-  restore_win(&switchwin, true);
-
-  redraw_later(win, UPD_VALID);
-  win->w_redr_status = true;
+  if (win) {
+    win->w_cursor.lnum = (linenr_T)row;
+    win->w_cursor.col = (colnr_T)col;
+    win->w_cursor.coladd = 0;
+    // When column is out of range silently correct it.
+    check_cursor_col(win);
+
+    // Make sure we stick in this column.
+    win->w_set_curswant = true;
+
+    // make sure cursor is in visible range and
+    // cursorcolumn and cursorline are updated even if win != curwin
+    switch_win(&switchwin, win, NULL, true);
+    update_topline(curwin);
+    validate_cursor(curwin);
+    if (opts->flush) {
+      setcursor_mayforce(true);
+    }
+    restore_win(&switchwin, true);
+    redraw_later(win, UPD_VALID);
+    win->w_redr_status = true;
+  } else {  // screen
+    ui_grid_cursor_goto(default_grid.handle, (int)row - 1, (int)col);
+    skip_cursor_update = true;
```

Comment:
just my immediate reaction, but this part seems to be _very_ different functionality, to the extent it should be a separate API call altogether.

Reply:
I see. Like I said I didn't even have a use case for it.
I just went "no opts array" -> nvim_set_cursor() -> add 2 lines for screen relative cursor.
If it isn't wanted perhaps this should just be nvim_set_cursor2()?